packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -148,12 +168,38 @@ export class ChainService implements ChainServiceInterface {
       to: assetHolderAddress,
       value: isEthFunding ? arg.amount : undefined,
     };
-    return this.sendTransaction(depositRequest);
+
+    const tx = await this.sendTransaction(depositRequest);
+
+    if (!tx) throw new Error('fundChannel failed to submit deposit transaction');
+
+    this.logger.info(
+      {
+        channelId: arg.channelId,
+        assetHolderAddress,
+        tx: tx.hash,
+      },
+      'Finished funding channel'
+    );
+
+    return tx;
   }
 
   async concludeAndWithdraw(
     finalizationProof: SignedState[]
   ): Promise<providers.TransactionResponse | void> {
+    if (finalizationProof.length === 0) {
+      this.logger.warn('ChainService: concludeAndWithdraw was called with an empty array?');
+      return;
```

Comment:
We might want to throw an error?

Reply:
It's kind of a harmless situation, so I'm not sure if we want to

Reply:
Is there ever a valid scenario for calling `concludeAndWithdraw` with no finalization proof?

Reply:
a8bd7ea02